compiler/packages/babel-plugin-react-compiler/src/Inference/InferReferenceEffects.ts

```diff
@@ -1287,20 +1304,18 @@ function inferBlock(
         break;
       }
       case 'PropertyStore': {
-        const effect =
-          state.kind(instrValue.object).kind === ValueKind.Context
-            ? Effect.ConditionallyMutate
-            : Effect.Capture;
         state.referenceAndRecordEffects(
           freezeActions,
           instrValue.value,
-          effect,
+          Effect.Capture,
           ValueReason.Other,
         );
```

Comment:
Similar to the above hack for Store effects, a PropertyStore is always capturing its rvalue, never mutating it.

Reply:
This makes more sense, I was always kind of confused about the conditionally-mutate-if-its-a-context-variable logic